typescript/sdk/src/token/EvmERC20WarpRouteReader.ts

```diff
@@ -111,15 +181,45 @@ export class EvmERC20WarpRouteReader {
    * @param tokenAddress - The address of the token.
    * @returns A partial ERC20 metadata object containing the token name, symbol, total supply, and decimals.
    */
-  async fetchTokenMetadata(tokenAddress: Address): Promise<ERC20Metadata> {
-    const erc20 = ERC20__factory.connect(tokenAddress, this.provider);
-    const [name, symbol, totalSupply, decimals] = await Promise.all([
+  async fetchTokenMetadata(
+    type: DerivedTokenType,
+    tokenAddress: Address,
+  ): Promise<ERC20Metadata & { token?: string }> {
+    if (type === TokenType.collateral || type === TokenType.collateralVault) {
```

Comment:
can we pull this out into a constant somewhere
"collateral" types

Reply:
not sure if this adds much value. 

I believe all of the other collaterals checks are of the variety, config.type === collateral || config.type === collaterlaVault. Only time when the property `type` is checked is in the deployer, which is where i copied this from

Reply:
moved to `CollateralExtensions`